src/main/java/org/generationcp/middleware/service/impl/inventory/TransactionServiceImpl.java

```diff
@@ -192,36 +193,43 @@ public void depositLots(final Integer userId, final Set<Integer> lotIds, final L
 				.getByGids(lots.stream().map(ExtendedLotDto::getGid).collect(Collectors.toSet())).stream()
 				.collect(Collectors.toMap(a -> a.getGermplasm().getGid(), Function.identity()));
 
-		for (final ExtendedLotDto lotDto : lots) {
-			final Double amount = lotDepositRequestDto.getDepositsPerUnit().get(lotDto.getUnitName());
-			final Transaction transaction = new Transaction();
-			transaction.setStatus(transactionStatus.getIntValue());
-			transaction.setType(TransactionType.DEPOSIT.getId());
-			transaction.setLot(new Lot(lotDto.getLotId()));
-			transaction.setPersonId(userId);
-			transaction.setUserId(userId);
-			transaction.setTransactionDate(new Date());
-			transaction.setQuantity(amount);
-			transaction.setComments(lotDepositRequestDto.getNotes());
-			//Always zero for new transactions
-			transaction.setPreviousAmount(0D);
-			if (transactionStatus.equals(TransactionStatus.CONFIRMED)) {
-				transaction.setCommitmentDate(Util.getCurrentDateAsIntegerValue());
-			} else {
-				transaction.setCommitmentDate(0);
-			}
-			this.daoFactory.getTransactionDAO().save(transaction);
+		for (final ExtendedLotDto extendedLotDto : lots) {
+			final Double amount = lotDepositRequestDto.getDepositsPerUnit().get(extendedLotDto.getUnitName());
+			final Transaction transaction =
+				new Transaction(TransactionType.DEPOSIT, transactionStatus, userId, lotDepositRequestDto.getNotes(),
+					extendedLotDto.getLotId(),
+					amount);
+			daoFactory.getTransactionDAO().save(transaction);
 
 			if (lotDepositRequestDto.getSourceStudyId() != null) {
 				// Create experiment transaction records when lot and deposit are created in the context of study.
-				this.createExperimentTransaction(lotDto.getGid(), germplasmStudySourceMap, transaction,
+				this.createExperimentTransaction(extendedLotDto.getGid(), germplasmStudySourceMap, transaction,
 					ExperimentTransactionType.HARVESTING);
 			}
-
 		}
 
 	}
 
+	@Override
+	public void depositLots(final Integer userId, final Set<Integer> lotIds, final List<LotDepositDto> lotDepositDtoList,
+		final TransactionStatus transactionStatus) {
+
+		final LotsSearchDto lotsSearchDto = new LotsSearchDto();
+		lotsSearchDto.setLotIds(new ArrayList<>(lotIds));
+		final List<ExtendedLotDto> lots = this.daoFactory.getLotDao().searchLots(lotsSearchDto, null);
+		final Map<String, List<LotDepositDto>> lotDepositDtoMap = lotDepositDtoList.stream().collect(Collectors.groupingBy(LotDepositDto::getLotUID));
+
+		for (final ExtendedLotDto extendedLotDto : lots) {
```

Comment:
@cuenyad Having 2 for is not needed. Iterate lotDepositDtoList and have a map with LotUID as key